pkg/mutation/schema/schema_test.go

```diff
@@ -5,12 +5,10 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/open-policy-agent/frameworks/constraint/pkg/externaldata"
```

Comment:
It looks like external data is adding a new bit of schema information...

Since the keys referenced by external data can only be primitives, we cannot have subpaths of a mutator that references external data.

For instance, if we have:

Mutator A: set `spec.fields.foo` to the value returned by external data provider AssignFoo

Then:

Mutator B: set `spec.fields.foo.bar` is invalid.

We should update the schema validator to detect this conflict.

Reply:
Still working on addressing this comment but the latest commit should address the rest of the comments.